src/Orleans.Transactions/State/TransactionalState.cs

```diff
@@ -501,6 +482,38 @@ public override string ToString()
         {
             return $"{this.context.GrainInstance}.{this.config.StateName}";
         }
+
+        [Serializable]
+        private class Metadata
+        {
+            [JsonIgnore]
+            public TransactionalResourceVersion HighestVersion { get; set; }
+            public string HighestVersionString
+            {
+                get { return this.HighestVersion.ToString(); }
+                set { this.HighestVersion = (TransactionalResourceVersion.TryParse(value, out TransactionalResourceVersion version)) ? version : default(TransactionalResourceVersion); }
+            }
+
+            [JsonIgnore]
+            public TransactionalResourceVersion StableVersion { get; set; }
+            public string StableVersionString
+            {
+                get { return this.StableVersion.ToString(); }
+                set { this.StableVersion = (TransactionalResourceVersion.TryParse(value, out TransactionalResourceVersion version)) ? version : default(TransactionalResourceVersion); }
+            }
+
+            public long HighestReadTransactionId { get; set; }
+
+            public override string ToString()
+            {
+                return JsonConvert.SerializeObject(this);
```

Comment:
I'm a little scared of just using the default settings by using JsonConvert, since a user might modify the global defaults for JsonConvert. We should use `JsonSerializer.Create` and pass explicit settings (and then cache to a static field)